server/src/main/java/keywhiz/service/daos/SecretDAO.java

```diff
@@ -283,6 +286,35 @@ public boolean setExpiration(String name, Instant expiration) {
     return Optional.empty();
   }
 
+  /**
+   * @param names of secrets series to look up secrets by.
+   * @return Secrets matching input parameters.
+   */
+  public List<SecretSeriesAndContent> getSecretsByName(List<String> names) {
+    checkArgument(!names.isEmpty());
+
+    SecretContentDAO secretContentDAO = secretContentDAOFactory.using(dslContext.configuration());
+    SecretSeriesDAO secretSeriesDAO = secretSeriesDAOFactory.using(dslContext.configuration());
+
+    List<SecretSeries> multipleSeries = secretSeriesDAO.getMultipleSecretSeriesByName(names);
+
+    List<SecretSeriesAndContent> ret = new ArrayList<SecretSeriesAndContent>();
+
+    for (SecretSeries series : multipleSeries) {
+      if (series.currentVersion().isPresent()) {
+        long secretContentId = series.currentVersion().get();
+        Optional<SecretContent> secretContent =
+                secretContentDAO.getSecretContentById(secretContentId);
```

Comment:
wouldn't it make sense to also make this a batch query? (so `getSecretContentsByIds`?)

Reply:
Yes, I'll make myself a ticket to implement this later. Good call!